test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientTests/Tests/ClientEntityDescriptorTests.cs

```diff
@@ -0,0 +1,202 @@
+        {
+            {"People", new List<string>{"PersonID"}},
+            {"Orders", new List<string>{"OrderID"}},
+            {"ProductDetails", new List<string>{"ProductID", "ProductDetailID"}}
+        };
+
+        [Fact]
+        public void EntityDescriptor_LinksAndIdentity_AreNotNull_ForAllEntities()
+        {
+            foreach (var entitySetKeys in entitySetsList.Keys)
+            {
+                IQueryable iqueryableProperty = typeof(Container).GetProperty(entitySetKeys).GetValue(_context, null) as IQueryable;
+                foreach (var entity in iqueryableProperty)
+                {
+                    EntityDescriptor eDescriptor = _context.GetEntityDescriptor(entity);
+                    //Self link was not read
+                    Assert.NotNull(eDescriptor.SelfLink);
+                    //Edit link was not read
+                    Assert.NotNull(eDescriptor.EditLink);
+                    //Identity was not read
+                    Assert.NotNull(eDescriptor.Identity);
+                }
+            }
+        }
+
+        [Fact]
+        public void TopQuery_ExecutesSuccessfully()
+        {
+            var person = _context.People.Take(1);
+            Assert.Equal("http://localhost/odata/People?$top=1", person.ToString());
+            Assert.Single(person);
+        }
+
+        [Fact]
+        public void SkipOption_ExecutesSuccessfully()
+        {
+            var people = _context.People.Skip(1);
+            Assert.Equal("http://localhost/odata/People?$skip=1", people.ToString());
+            Assert.NotNull(people.FirstOrDefault().PersonID);
+        }
+
+        [Fact]
+        public void OrderByOption_ExecutesSuccessfully()
+        {
+            var people = from p in _context.People 
+                         orderby p.FirstName 
+                         select p;
+
+            Assert.Equal("http://localhost/odata/People?$orderby=FirstName", people.ToString());
+            Assert.NotNull(people.ToList());
+        }
+
+        [Fact]
+        public void OrderByDescendingOption_ExecutesSuccessfully()
+        {
+            var people = from p in _context.People
+                         orderby p.FirstName descending
+                         select p;
+
+            Assert.Equal("http://localhost/odata/People?$orderby=FirstName desc", people.ToString());
+
+            Assert.NotNull(people.FirstOrDefault());
+        }
+
+        [Fact]
+        public void GetDiscontinuedProducts()
+        {
+            var discontinuedProducts = from product in _context.Products
+                                       where product.Discontinued
+                                       select product;
+
+            Assert.Equal("http://localhost/odata/Products?$filter=Discontinued", discontinuedProducts.ToString());
+
+            Assert.NotNull(discontinuedProducts.FirstOrDefault());
+        }
+
+        [Fact]
+        public void GetInCirculationProducts()

```

Comment:
```suggestion
        public void FilterByNegatedBooleanProperty_ExecutesSuccessfully()
```
See: https://github.com/OData/odata.net/pull/3027/files#r1716435731